File: 0326.3的幂/0326-3的幂.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        def h(n):
            if n <= 0:
                return False
            if n == 1:
                return True
            if n%3 == 0:
                return h(n//3)
            return False
        return h(n)
        

File: 0326.3的幂/0326-3的幂.rs
impl Solution {
    pub fn is_power_of_three(n: i32) -> bool {
        let n: i64 = n as i64;
        for i in 0.. {
            if n == 3_i64.pow(i as u32) {
                return true;
            }
            if 3_i64.pow(i as u32) > n {
                return false;
            }
        }
        return false;
    }
}